libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -459,6 +455,18 @@ size_t WiFiClient::write(Stream &stream)
 
 int WiFiClient::read(uint8_t *buf, size_t size)
 {
+    if(_lastReadTimeout != _timeout){
+        if(fd() >= 0){
+            struct timeval timeout_tv;
+            timeout_tv.tv_sec = _timeout/1000;
+            timeout_tv.tv_usec = 0;
```

Comment:
should this be `_timeout % 1000`? (To support timeouts less than one second)

Reply:
Should it be implemented this way to use `tv_sec` and `tv_usec` at the same time?

```
timeout_tv.tv_sec = _timeout / 1000;
timeout_tv.tv_usec = (_timeout % 1000) * 1000;
```

For 1500ms `_timeout` that will be `timeout_tv.tv_sec = 1` and `timeout_tv.tv_usec = 500000`

Reply:
Edited, now the seconds are not rounded and timeout can be less than 1 second